readinglist/views/batch.py

```diff
@@ -0,0 +1,165 @@
+import json
+
+import colander
+from cornice import Service
+from pyramid.request import Request
+from pyramid import httpexceptions
+import six
+from six.moves.urllib import parse as urlparse
+
+from readinglist import logger
+from readinglist import errors
+
+
+valid_http_method = colander.OneOf(('GET', 'HEAD', 'DELETE', 'TRACE',
+                                    'POST', 'PUT', 'PATCH'))
+
+
+def string_values(node, cstruct):
+    """Validate that a ``colander.Mapping`` only has strings in its values.
+
+    :warning:
+
+        Should be associated to a ``colander.Mapping`` schema node.
+    """
+    are_strings = [isinstance(v, six.string_types) for v in cstruct.values()]
+    if not all(are_strings):
+        error_msg = '%s contains non string value' % cstruct
+        raise colander.Invalid(node, error_msg)
+
+
+class BatchRequestSchema(colander.MappingSchema):
+    method = colander.SchemaNode(colander.String(),
+                                 validator=valid_http_method,
+                                 missing=colander.drop)
+    path = colander.SchemaNode(colander.String(),
+                               validator=colander.Regex('^/'))
+    headers = colander.SchemaNode(colander.Mapping(unknown='preserve'),
+                                  validator=string_values,
+                                  missing=colander.drop)
+    body = colander.SchemaNode(colander.Mapping(unknown='preserve'),
+                               missing=colander.drop)
+
+
+class BatchPayloadSchema(colander.MappingSchema):
+    requests = colander.SchemaNode(colander.Sequence(),
+                                   BatchRequestSchema())
+
+    def deserialize(self, cstruct=colander.null):
+        """Preprocess received data
+        """
+        if cstruct is colander.null:
+            return colander.null
+
+        # See if defaults was specified in payload
```

Comment:
At least, we should rephrase the comment here: "See if default arguments were specified"